src/Translator.php

```diff
@@ -0,0 +1,512 @@
+            if (preg_match('#^(?<dir>.+)\\/(?<file>[^\\/]+)\\.yml$#', $source, $matches)) {
+                return $modulePath . '/' . $matches['dir'];
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Gets the directories of the JS lang folder.
+     */
+    private function getJSLangDirectories(string $modulePath): array
+    {
+        $sources = $this->getTransifexSources($modulePath);
+        $dirs = [];
+        foreach ($sources as $source) {
+            // Strip out /src/ dir and trailing file.js
+            if (preg_match('#^(?<dir>.+)\\/src\\/(?<file>[^\\/]+)\\.js(on)?$#', $source, $matches)) {
+                $dirs[] = $modulePath . '/' . $matches['dir'];
+            }
+        }
+        return $dirs;
+    }
+
+    /**
+     * Get list of transifex source files. E.g. lang/en.yml
+     */
+    private function getTransifexSources(string $modulePath): array
+    {
+        $path = "$modulePath/.tx/config";
+        $content = file_get_contents($path);
+        $sources = [];
+        foreach (preg_split('~\R~u', $content) as $line) {
+            if (preg_match('#source_file\s=\s(?<path>\S+)#', $line, $matches)) {
+                $sources[] = $matches['path'];
+            }
+        }
+        return $sources;
+    }
+
+    /**
+     * Generate javascript for all modules
+     */
+    private function generateJavascript(): void
+    {
+        $this->log("Generating javascript locale files");
+        // Check which paths in each module require processing
+        $count = 0;
+        foreach ($this->modulePaths as $modulePath) {
+            $jsPaths = $this->getJSLangDirectories($modulePath);
+            foreach ((array)$jsPaths as $jsPath) {
+                $count += $this->generateJavascriptInDirectory($modulePath, $jsPath);
+            }
+        }
+        $this->log("Finished generating {$count} files");
+    }
+
+    /**
+     * Process all javascript in a given path
+     */
+    private function generateJavascriptInDirectory(string $modulePath, string $jsPath): int
+    {
+        $count = 0;
+        foreach (glob("{$jsPath}/src/*.js*") as $sourceFile) {
+            // re-encode contents to ensure they're consistently formatted
+            $sourceContents = $this->jsonEncode($this->jsonDecode(file_get_contents($sourceFile)));
+            $locale = preg_replace('/\.js.*$/', '', basename($sourceFile));
+            $targetFile = dirname(dirname($sourceFile)) . '/' . $locale . '.js';
+            $this->log("Generating file {$targetFile}");
+            $targetContents = <<<TEMPLATE
+            // This file was generated by silverstripe/cow from %FILE%.
+            // See https://github.com/silverstripe/cow for details
+            if (typeof(ss) === 'undefined' || typeof(ss.i18n) === 'undefined') {
+              if (typeof(console) !== 'undefined') { // eslint-disable-line no-console
+                console.error('Class ss.i18n not defined');  // eslint-disable-line no-console
```

Comment:
This indentation matches what cow used to do

Reply:
https://github.com/silverstripe/cow/blob/5682d6d29cca06b2f974f09273bd45a1c5c82c1d/src/Utility/Translations.php#L56